From f44ce2ef17449c253401e571be3211fce7401caa Mon Sep 17 00:00:00 2001
From: kostap <kostap@marvell.com>
Date: Thu, 19 Jun 2014 15:25:22 +0300
Subject: [PATCH 1789/1825] fix: msys: Correct the SGMII ports definitions and
 ports quantity

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 9ec37e716aee9106303389cdf8c1cc3858191b03

    - Fix the amount and PHY addresses of SGMII ports in AC3 DB configuration structures
    - Fix the amount of SGMII ports for BC2 boards

Change-Id: I8f8647a26ec3297f7661ca71c596bd1f9650b35c
Signed-off-by: kostap <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/8625
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9170
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../msys_family/boardEnv/mvBoardEnvSpec.c          |    4 ++--
 .../mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c   |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-msys/msys_family/boardEnv/mvBoardEnvSpec.c b/arch/arm/mach-msys/msys_family/boardEnv/mvBoardEnvSpec.c
index efbbbaa..fb14d3c 100644
--- a/arch/arm/mach-msys/msys_family/boardEnv/mvBoardEnvSpec.c
+++ b/arch/arm/mach-msys/msys_family/boardEnv/mvBoardEnvSpec.c
@@ -591,8 +591,8 @@ MV_BOARD_TWSI_INFO	db_dx_ac3InfoBoardTwsiDev[] = {
 
 MV_BOARD_MAC_INFO db_dx_ac3InfoBoardMacInfo[] = {
 	/* {{MV_BOARD_MAC_SPEED	boardMacSpeed, MV_U8 boardEthSmiAddr}} */
-	{BOARD_MAC_SPEED_AUTO, 0x0 },
-	{BOARD_MAC_SPEED_AUTO, 0x1 },
+	{BOARD_MAC_SPEED_AUTO, -1 },
+	{BOARD_MAC_SPEED_AUTO, -1 }
 };
 
 MV_BOARD_MODULE_TYPE_INFO db_dx_ac3InfoBoardModTypeInfo[] = {
diff --git a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
index 7c71f5e..a5d792f 100644
--- a/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-msys/msys_family/ctrlEnv/mvCtrlEnvLib.c
@@ -104,7 +104,7 @@ MV_UNIT_ID mvCtrlSocUnitNums[MAX_UNITS_ID][MV_MSYS_INDEX_MAX] = {
 /*			    BC2		AC3 */
 /* DRAM_UNIT_ID         */ { 1,		1, },
 /* PEX_UNIT_ID          */ { 1,		1, },
-/* ETH_GIG_UNIT_ID      */ { 2,		2, },
+/* ETH_GIG_UNIT_ID      */ { 1,		2, },
 /* XOR_UNIT_ID          */ { 1,		1, },
 /* UART_UNIT_ID         */ { 2,		2, },
 /* SPI_UNIT_ID          */ { 2,		1, },
-- 
1.7.5.4

